iterative/resource_runner.go

```diff
@@ -172,6 +176,43 @@ func resourceRunnerCreate(ctx context.Context, d *schema.ResourceData, m interfa
 		diags = resourceMachineCreate(ctx, d, m)
 	}
 
+	if diags.HasError() {
+		return diags
+	}
+
+	machineUser := "ubuntu"
+	machineKey := d.Get("ssh_private").(string)
+	machineAddress := net.JoinHostPort(d.Get("instance_ip").(string), "22")
+	machineLogCommand := "journalctl --unit cml"
+
+	var logError error
+	var logEvents string
+	err = resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		logEvents, logError = utils.RunCommand(machineLogCommand, 10*time.Second, machineAddress, machineUser, machineKey)
+		log.Printf("[DEBUG] Collected log events: %#v", logEvents)
+		log.Printf("[DEBUG] Connection errors: %#v", logError)
+
+		if logError != nil {
+			return resource.RetryableError(fmt.Errorf("Waiting for the machine to accept connections... %s", logError))
+		} else if !utils.IsReady(logEvents) {
+			return resource.RetryableError(fmt.Errorf("Waiting for the runner to be ready... %s", logError))
+		} else {
```

Comment:
we are missing one of the biggest needs here that is also log the error messages that may occur during the startup.
I.E

#### wrong token
```tf
resource "iterative_cml_runner" "runner-gh" {
    token = "WRONG_TOKEN"
    repo = "https://gitlab.com/iterative.ai/cml"
    driver = "gitlab"
    labels = "cml-runner"

    idle_timeout = 0
    cloud = "aws"
    region = "us-west"
    instance_type = "t2.micro"
    #instance_gpu = "tesla"

    spot = true
    # spot_price = 0.0070
}
```

to have this case totally solved https://github.com/iterative/cml/issues/413 should be solved first.
But maybe at least it does not get hung forever, so even that this can not be totally finished at least a timeout must proceed.

Reply:
e62b5513c9c4a1dc71495a1d7ca62900be562460 Timeouts defaulted to 20 minutes, but now they can be easily configured with a nested block.

```hcl
resource "iterative_cml_runner" "runner" {
  ···
  timeouts {
    create = "1m"
    delete = "1m"
    update = "1m"
  }
}
```

Reply:
Im not able to setup the timeout as you specify here

Reply:
@DavidGOrtega, it works on my computer ™

* macOS Catalina 10.15.7 (19H114) 
* Go 1.16.2 darwin/amd64
* Terraform v0.14.6

```hcl
resource "iterative_cml_runner" "runner" {
    repo              = "https://github.com/···"
    driver            = "github"
    labels            = "cml"
    idle_timeout      = "3600"
    name              = "debug-runner-2"
    cloud             = "aws"
    region            = "us-west"
    instance_type     = "t2.micro"
    timeouts {
        create = "4m"
        delete = "4m"
        update = "4m"
    }
}
```

Reply:
Are you experiencing a syntax error, or it's just silently ignoring them?